### PR TITLE
roachtest: fix rebalance/by-load/*/mixed-version shared process tests

### LINKED ISSUE #129464: roachtest: rebalance/by-load/replicas/mixed-version failed
roachtest.rebalance/by-load/replicas/mixed-version [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/16573719?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/16573719?buildTab=artifacts#/rebalance/by-load/replicas/mixed-version) on master @ [18c4e23306bb99cc28c83bb3e5726f6f4279b798](https://github.com/cockroachdb/cockroach/commits/18c4e23306bb99cc28c83bb3e5726f6f4279b798):

```
(mixedversion.go:710).Run: mixed-version test failure while running step 15 (run "rebalance load run"): CPU not evenly balanced after timeout: outside bounds mean=99.2 tolerance=20.0% (±19.8) bounds=[79.3, 119.0]
	below  = []
	within = [s1: 85 (-14.2%), s2: 86 (-12.4%), s3: 88 (-10.5%), s5: 80 (-18.6%), s6: 85 (-13.6%)]
	above  = [s4: 167 (+69.3%)]
test artifacts and logs in: /artifacts/rebalance/by-load/replicas/mixed-version/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=azure</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_runtimeAssertionsBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*rebalance/by-load/replicas/mixed-version.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

Jira issue: CRDB-41554

### LINKED ISSUE #129962: roachtest: rebalance/by-load/replicas/mixed-version failed
**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar issue without the B-runtime-assertions-enabled label. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.rebalance/by-load/replicas/mixed-version [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16695535?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16695535?buildTab=artifacts#/rebalance/by-load/replicas/mixed-version) on master @ [8551145a0c99c4c95a28ec470e699d0c20ca97ab](https://github.com/cockroachdb/cockroach/commits/8551145a0c99c4c95a28ec470e699d0c20ca97ab):

```
(mixedversion.go:710).Run: mixed-version test failure while running step 38 (run "rebalance load run"): CPU not evenly balanced after timeout: outside bounds mean=113.4 tolerance=20.0% (±22.7) bounds=[90.7, 136.0]
	below  = [s1: 87 (-23.0%), s2: 63 (-44.3%), s5: 58 (-48.0%)]
	within = [s6: 134 (+19.0%)]
	above  = [s3: 170 (+50.6%), s4: 165 (+45.6%)]
test artifacts and logs in: /artifacts/rebalance/by-load/replicas/mixed-version/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_runtimeAssertionsBuild=true</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16695535/rebalance-by-load-replicas-mixed-version/1725105798904/1725107498861)

</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*rebalance/by-load/replicas/mixed-version.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

Jira issue: CRDB-41804

### LINKED ISSUE #131274: roachtest: rebalance/by-load/leases/mixed-version failed
**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar issue without the B-runtime-assertions-enabled label. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.rebalance/by-load/leases/mixed-version [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17010157?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17010157?buildTab=artifacts#/rebalance/by-load/leases/mixed-version) on master @ [a055dec86299452f1951416c16f5938e523f8559](https://github.com/cockroachdb/cockroach/commits/a055dec86299452f1951416c16f5938e523f8559):

```
(mixedversion.go:720).Run: mixed-version test failure while running step 12 (run "rebalance load run"): CPU not evenly balanced after timeout: outside bounds mean=117.1 tolerance=20.0% (±23.4) bounds=[93.7, 140.5]
	below  = [s2: 90 (-22.3%), s3: 85 (-27.0%)]
	within = []
	above  = [s1: 174 (+49.3%)]
test artifacts and logs in: /artifacts/rebalance/by-load/leases/mixed-version/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_runtimeAssertionsBuild=true</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-17010157/rebalance-by-load-leases-mixed-version/1727159228632/1727160042515)

</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*rebalance/by-load/leases/mixed-version.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

Jira issue: CRDB-42466

### PR DESCRIPTION
**Pull request by kvoli**
In https://github.com/cockroachdb/cockroach/pull/129117, `rebalance/by-load/*/mixed-version` roachtest had
shared-process multi-tenancy introduced, which would occasionally cause
these tests to fail erroneously.

The cause of all the failures was identical, CPU utilization of some
nodes which couldn't have been possible, > 100%, e.g.,

```
CPU not evenly balanced after timeout: outside bounds mean=102.5 tolerance=20.0% (±20.5) bounds=[82.0, 123.0]
below  = [s3: 81 (-20.7%), s5: 65 (-36.5%)]
within = [s2: 116 (+14.0%), s4: 92 (-9.7%), s6: 88 (-13.2%)]
above  = [s1: 170 (+66.1%)]
```

As the query would aggregate every tenant's timeseries data on a given
node, instead of only the system tenant.

Update the timeseries utility used to query the CPU to also take in a
`TenantID` parameter, which is then used to query only the system
tenant.

Fixes: https://github.com/cockroachdb/cockroach/issues/129962
Fixes: https://github.com/cockroachdb/cockroach/issues/131274
Fixes: https://github.com/cockroachdb/cockroach/issues/129464
Release note: None

### CODE REVIEW

**Review by stevendanna — APPROVED** [review]

**Review by tbg — APPROVED** [review]

File: pkg/cmd/roachtest/tests/rebalance_load.go [hunk 1]
```diff
@@ -367,6 +369,14 @@ func makeStoreCPUFn(
 			}
 			// Take the latest CPU data point only.
 			cpu := result.Datapoints[len(result.Datapoints)-1].Value
+			// The datapoint is a float representing a percentage in [0,1.0]. Assert
+			// as much to avoid any surprises.
+			if cpu < 0 || cpu > 1 {
```
Comment (kvoli) [inline pkg/cmd/roachtest/tests/rebalance_load.go:374]: Fixed this on latest push: should be > 1, as we multiply [0,1] to get [0,100] below.

**Comment by kvoli** [comment]
TYFTRs!

**Comment by kvoli** [comment]
bors r=stevendanna,tbg

**Comment by kvoli** [comment]
blathers backport 24.2

**Comment by kvoli** [comment]
blathers backport 24.1

**Comment by kvoli** [comment]
blathers backport 23.2

**Pull request merged**